File: design-patterns/go/chain_of_responsibility/cache/redis_cache_handler.go
// redis_cache_handler.go

package main

import "fmt"

type RedisCacheHandler struct {
	nextCacheHandler CacheHandler
}

func NewRedisCacheHandler(nextCacheHandler CacheHandler) (redisCacheHandler *RedisCacheHandler) {
	redisCacheHandler = &RedisCacheHandler{}
	redisCacheHandler.nextCacheHandler = nextCacheHandler
	return
}

func (redisCacheHandler *RedisCacheHandler) HandleRequest(data Data) {
	if data.GetDataType() == DATA && len(data.GetValue()) <= 1024 {
		fmt.Printf("Caching data '%v' in Redis\n", data.GetKey())
	} else if redisCacheHandler.nextCacheHandler != nil {
		redisCacheHandler.nextCacheHandler.HandleRequest(data)
	}
}

File: design-patterns/php/chainofresponsibility/cache/RedisCacheHandler.php
<?php
// RedisCacheHandler.php

namespace BigBoxCode\DesignPattern\ChainOfResponsibility\Cache;


class RedisCacheHandler extends CacheHandler {
    public function __construct(?CacheHandler $nextCacheHandler) {
        parent::__construct($nextCacheHandler);
    }

    public function handleRequest(Data $data): void {
        if ($data->getType() == DATA_TYPE::DATA && strlen($data->getData()) <= 1024) {
            echo "Caching file '" . $data->getKey() . "' in Redis\n";
        } else if ($this->nextCacheHandler != null) {
            $this->nextCacheHandler->handleRequest($data);
        }
    }
}
